File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        // Step 1: If lengths differ, not an anagram
        if (s.length() != t.length()) return false;

        // Step 2: Count character frequencies
        int[] count = new int[26]; // for lowercase letters only

        for (int i = 0; i < s.length(); i++) {
            count[s.charAt(i) - 'a']++;
            count[t.charAt(i) - 'a']--;
        }

        // Step 3: Check if all counts are zero
        for (int c : count) {
            if (c != 0) return false;
        }

        return true;
    }
}

File: 0242-valid-anagram/0242-valid-anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    if(s.length !== t.length ) return false;

    let map = new Map();

    for(let i = 0; i < s.length; i++){
        map.set(s[i], (map.get(s[i]) || 0) +1)
        map.set(t[i], (map.get(t[i]) || 0) -1)
    }

    for(let val of map.values()){
        if(val !== 0 ) return false;
    }
    return true;
};